### PR TITLE
Logic actions

### PR DESCRIPTION
**Pull request by hestonhoffman**
See also:

- #19097
- #19096

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

File: content/en/service_management/workflows/actions_catalog/logic_actions.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+## Sleep
+
+The **Sleep** action pauses the execution of the workflow for a specified duration. Select a predefined duration from the **Duration** drop-down, or enter a custom variable in seconds.
+
+## Iteration
+
+The **For each** action allows you to execute a specific action iteratively for each element in a given input list. It enables you to automate repetitive tasks by applying the same action to multiple items within a list.
+
+The action is made up of the For each step and an inner step that you intend to perform iteratively. The output of the a For each step is an array of outputs from the inner step.
```
Comment (urseberry) [inline content/en/service_management/workflows/actions_catalog/logic_actions.md:29]:
```suggestion
The action is made up of the For each step and an inner step that you intend to perform iteratively. The output of a For each step is an array of outputs from the inner step.
```

File: content/en/service_management/workflows/actions_catalog/logic_actions.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+For example, using For each together with a CloudFlare action, you could iterate over and block a list of IP addresses. In this scenario, the CloudFlare action is placed inside a For each step and the list of IP addresses is added to the For each step as an **Input list**. When the workflow runs, the CloudFlare step repeats for each value in the list, accessing the current IP value and blocking it.
```
Comment (urseberry) [inline content/en/service_management/workflows/actions_catalog/logic_actions.md:31]: Consider rewording the "In this scenario..." sentence into active voice.

File: content/en/service_management/workflows/actions_catalog/logic_actions.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+1. Search for and select the **For each** step.
+1. To select an inner action to repeat, drag an existing step from your canvas into the **For each** step, or click the plus (**+**) icon inside the **For each** step and make a selection from the action catalog.
+1. Click the **For each** step and enter an **Input list** for the step to iterate over.
+1. Toggle **Continue on error** if you want the action to continue down the list of remaining values when an error is encountered.
+1. Configure the inner action. To access the current value in the input list, use the **{{Current.value}}** variable. To access the index of the current value, use **{{Current.index}}**.
+1. **Save** and **Run** the action.
+
+When a run completes, the workflow enters **Debug** mode. Select the For each step to see:
+* The output of each iteration
+* The number of completed and failed iterations.
```
Comment (urseberry) [inline content/en/service_management/workflows/actions_catalog/logic_actions.md:44]:
```suggestion
* The number of completed and failed iterations
```

File: content/en/service_management/workflows/actions_catalog/logic_actions.md [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+You can delete the inner step by:
+- Clicking the inner step and selecting **Delete**.
+- Clicking the For each step and selecting **Clear**.
```
Comment (urseberry) [inline content/en/service_management/workflows/actions_catalog/logic_actions.md:48]:
```suggestion
- Clicking the inner step and selecting **Delete**
- Clicking the For each step and selecting **Clear**
```

File: content/en/service_management/workflows/actions_catalog/logic_actions.md [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+The **Sleep** action pauses the execution of the workflow for a specified duration. Select a predefined duration from the **Duration** drop-down, or enter a custom variable in seconds.
+
+## Iteration
+
+The **For each** action allows you to execute a specific action iteratively for each element in a given input list. It enables you to automate repetitive tasks by applying the same action to multiple items within a list.
+
+The action is made up of the For each step and an inner step that you intend to perform iteratively. The output of a For each step is an array of outputs from the inner step.
+
+For example, using For each together with a CloudFlare action, you could iterate over and block a list of IP addresses. In this scenario, you would add the list of IP addresses as an **Input list** in the For each step. Next, you would add a CloudFlare action as an inner step and configure it to block the current value in the iteration loop. When the workflow runs, the CloudFlare step would repeat for each value in the list, accessing the current IP value and blocking it.
```
Comment (urseberry) [inline content/en/service_management/workflows/actions_catalog/logic_actions.md:31]:
```suggestion
For example, using For each together with a CloudFlare action, you can iterate over and block a list of IP addresses. In this scenario, you add the list of IP addresses as an **Input list** in the For each step. Next, add a CloudFlare action as an inner step and configure it to block the current value in the iteration loop. When the workflow runs, the CloudFlare step repeats for each value in the list, accessing the current IP value and blocking it.
```

**Comment by hestonhoffman** [comment]
Thanks @urseberry ❤️

**Pull request merged**